compiler/rustc_arena/src/lib.rs

```diff
@@ -214,7 +214,9 @@ impl<T> TypedArena<T> {
             // We need the element size to convert chunk sizes (ranging from
             // PAGE to HUGE_PAGE bytes) to element counts.
             let elem_size = cmp::max(1, mem::size_of::<T>());
-            let mut chunks = self.chunks.borrow_mut();
+
+            let chunks = &mut *self.chunks.as_ptr();
```

Comment:
We should use UnsafeCell or RefCell; if we're not going to use Cell's safe API then there's no reason for us to use Cell, right?